embedder/embedder.go

```diff
@@ -75,36 +74,38 @@ type FlutterEngine struct {
 	// Engine arguments
 	AssetsPath  string
 	IcuDataPath string
+
+	// AOT ELF snopshot path
+	// only required for AOT app.
+	ElfSnapshotPath string
+	// aotDataSource   C.FlutterEngineAOTData
+}
+
+// GoError convert a FlutterEngineResult to a golang readable error
+func (res Result) GoError(caller string) error {
```

Comment:
Instead of taking the caller name as argument to GoError(), we could use runtime information to obtain the caller name.
Example usage:
```go
    pc, _, _, ok := runtime.Caller(1)
    callerDetails := runtime.FuncForPC(pc)
    if ok && callerDetails != nil {
        fmt.Printf("called from %s\n", callerDetails.Name())
    }
```

I'm not sure what scenario's we could encounter where `ok` is false. It will be false when too many frames are skipped, but for value `1` that can never be the case in Go.

Reply:
Nevermind, I think this won't work when debug symbols are stripped, which hover tells Go to do when building a release build. (`-ldflags="-s -w"`)

Reply:
I really like this new `(Result).GoError()` by the way, a lot better to pass errors arround instead of Result. :+1: :sunglasses: